### DISCUSSION TITLE
Include push_bulk facility

### DISCUSSION THREAD

**Discussion by rainerborene** [sidekiq/sidekiq#4999] [Show and tell]
This is a very common use-case for Sidekiq and could be a built-in feature.  There should be no need for external gems, monkey patch or concerns to do basic stuff like that.

```ruby
# frozen_string_literal: true

module Bulkable
  extend ActiveSupport::Concern

  class_methods do
    def push_bulk!(items, queue, &block)
      if block
        args = items.map(&block)
      else
        args = items.map { |el| [el] }
      end

      Sidekiq::Client.push_bulk("class" => self, "args" => args, "queue" => queue)
    end

    def push_bulk(items, queue: :default, limit: 10_000, &block)
      job_ids = items.each_slice(limit).map do |group|
        push_bulk!(group, queue, &block)
      end

      job_ids.flatten
    end
  end
end
```

**Comment by mperham** [comment]
That's cool that you built an abstraction layer but I think the `push_bulk` API is easy enough to use directly.

**Comment by wilsonsilva** [comment]
@mperham What are the limits of push_bulk? For example, how many integers can one push per call?

**Reply by wilsonsilva** [reply]
Just read a comment in the method's implementation saying it is not recommended pushing more than 1000.
